cmd/agentbasedinstaller/host_config.go

```diff
@@ -192,10 +193,28 @@ func LoadHostConfigs(hostConfigDir string) (HostConfigs, error) {
 			}
 		}
 
-		configs = append(configs, &hostConfig{
-			configDir:    hostPath,
-			macAddresses: addresses,
-		})
+		addHostConfig := false
+		if workflowType == "addnodes" {
+			// In the addnodes workflow, the only host config we want to load is the
+			// current host's. Multiple HostConfigs could exist in hostConfigDir
+			// if multiple day-2 nodes are being added using the same day-2 ISO.
+			// Filter otu the other HostConfig entries because each day-2
+			// node is added in isolation using their own internal assisted-service
+			// instance.
+			addHostConfig, err = currentHostHasMACAddress(addresses)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			addHostConfig = true
```

Comment:
nit: to simplify a little bit the loop and drop the bool var, if workflowType is "addnodes" and the current entry does not match the current host, a `continue` before adding the `hostConfig` to `configs` should be enough